BrainPortal/cbrain_plugins/cbrain-plugins-base/userfiles/single_file/single_file.rb

```diff
@@ -80,4 +80,25 @@ def gzip_content(operation = :compress) #:nodoc:
     self.sync_to_provider
   end
 
+  # compute file md5 hash sum (useful to avoid tampering, e.g. with task archive)
+  def cache_compute_md5sum
+    md5command =
+        case CBRAIN::System_Uname
+        when /Linux/i
+          "md5sum"
+        when /Darwin/i
+          "md5"
+        else
+          "md5sum"
+        end
+    self.sync_to_cache
+
+    content_path = self&.cache_full_path
+    cb_error("Can't find cache path for file #{self.name}") if content_path.blank?
+    md5 = IO.popen("#{md5command} < #{content_path.to_s.bash_escape}","r") { |fh| fh.read }
```

Comment:
The computation code has a flaw. if the command's output is exactly 32 characters and not a MD5, the method will return those 32 characters:

* md5 is the raw output of the command
* it is present? but doesn't match the regex, so the assignment from last_match is not executed and md5 retains its original content
* thecondition on the cb_error thinks all is ok

Reply:
Solution: use a different variable name for the command's output VS the md5 extracted from the match

Reply:
done